RecoLocalMuon/GEMSegment/plugins/ME0SegAlgo.cc

```diff
@@ -0,0 +1,396 @@
+/**
+ * \file ME0SegAlgo.cc
+ *  based on CSCSegAlgoST.cc
+ * 
+ *  \authors: Marcello Maggi, Jason Lee
+ */
+ 
+#include "ME0SegAlgo.h"
+#include "MuonSegFit.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+#include "CommonTools/Statistics/interface/ChiSquaredProbability.h"
+
+#include <algorithm>
+#include <cmath>
+#include <iostream>
+#include <string>
+
+/* Constructor
+ *
+ */
+ME0SegAlgo::ME0SegAlgo(const edm::ParameterSet& ps) : ME0SegmentAlgorithm(ps), myName("ME0SegAlgo")
```

Comment:
the implementation looks like a copy of ME0SegAlgoMM with the same interface and ~95% the same implementation.
Please edit ME0SegAlgoMM instead, to deliver the slight change in the algorithm with a configuration option.